packages/web-components/src/components/ic-toggle-button-group/ic-toggle-button-group.tsx

```diff
@@ -0,0 +1,294 @@
+   */
+  @Prop() selectType?: IcSelectTypes = "single";
+
+  /**
+   * The size of the toggle buttons to be displayed. This does not affect the font size of the accessible label.
+   */
+  @Prop() size?: IcSizes = "default";
+
+  /**
+   * The variant of the toggle button.
+   */
+  @Prop({ reflect: true }) variant: "default" | "icon" = "default";
+
+  /**
+   * Emitted when a toggle button is selected.
+   */
+  @Event() icChange: EventEmitter<IcChangeEventDetail>;
+
+  @Listen("icToggleChecked")
+  selectHandler(ev: CustomEvent, tabTarget?: HTMLIcToggleButtonElement): void {
+    const allToggles = this.getAllToggleButtons();
+    let clickedToggle = ev.target as HTMLIcToggleButtonElement;
+    // tabTarget used in proxySelectHandler
+    tabTarget && tabTarget.focus();
+    if (this.selectType === "single") {
+      if (!clickedToggle) {
+        clickedToggle = tabTarget;
+      }
+      allToggles.forEach((el) => {
+        if (el.id !== clickedToggle.id && el.toggleChecked) {
+          el.toggleChecked = false;
+        }
+      });
+
+      this.icChange.emit({
+        toggleChecked: ev.detail.toggleChecked,
+        selectedOption: clickedToggle,
+      });
+    } else {
+      const toggledOptions = this.getAllToggleButtons().filter(
+        (el) => el.toggleChecked && !el.disabled
+      );
+
+      this.icChange.emit({
+        toggleChecked: toggledOptions.map((opt) => opt.toggleChecked),
+        toggledOptions: toggledOptions.map((opt) => ({
+          toggleButton: opt,
+        })),
+        selectedOption: clickedToggle,
+      });
+    }
+  }
+
+  componentWillLoad(): void {
+    this.selectType === "multi" && (this.selectMethod = "manual");
+    this.selectMethod === "auto" && this.selectType === "single";
+    document.addEventListener("keydown", this.keyListener);
+  }
+
+  componentDidLoad(): void {
+    this.getAllToggleButtons().forEach((el, i) => {
+      this.setSlottedAria(el);
+      el.size = this.size;
+      el.loading = this.loading;
+      el.iconPlacement = this.iconPlacement;
+      el.disabled ? null : (el.disabled = this.disabled);
+      el.appearance = this.appearance;
+      el.variant = this.variant;
+      el.fullWidth = this.fullWidth;
+      el.id = i.toString();
+      el.tabIndex = -1;
+      el.addEventListener("keydown", (ev) => {
+        this.handleKeyDown(ev);
+      });
+      el.classList.add("expand-toggle-group-child");
+    });
+  }
+
+  disconnectedCallback(): void {
+    document.removeEventListener("keydown", this.keyListener);
+  }
+
+  private keyListener = (ev: KeyboardEvent) => {
+    this.lastKeyPressed = {
+      key: ev.key,
+      shift: ev.shiftKey,
+    };
+  };
+
+  private setSlottedAria = (el: HTMLIcToggleButtonElement) => {
+    const btn = el.shadowRoot
+      .querySelector("ic-button")
+      .shadowRoot.querySelector("button") as HTMLButtonElement;
+    let aria = btn.getAttribute("aria-label");
+    aria += ", ";
+    aria += this.accessibleLabel;
```

Comment:
Adding the group name to each button's aria label feels like bad practice. 
I want to say this could be achieved better by adding a parent type role (`role="group"` or something) to the toggle button group and a child type role (`role="item"`) to the buttons.

Reply:
Please see internal comment regarding this